src/exla/exla.cc

```diff
@@ -439,14 +367,15 @@ ERL_NIF_TERM compile(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]){
   if(!exla::get_span(env, argv[1], argument_layouts)) return enif_make_badarg(env);
   if(!exla::get_options(env, argv[2], options)) return enif_make_badarg(env);
 
-  xla::StatusOr<std::vector<std::unique_ptr<xla::LocalExecutable>>> exec_status = xla_objects->client->Compile(*computation, argument_layouts, options);
-  // TODO: Handle this gracefully.
-  std::vector<std::unique_ptr<xla::LocalExecutable>> executables = exec_status.ConsumeValueOrDie();
+  EXLA_ASSIGN_OR_RETURN(std::vector<std::unique_ptr<xla::LocalExecutable>> executables,
+                        xla_objects->client->Compile(*computation, argument_layouts, options), env);
+
   ERL_NIF_TERM exec_refs[executables.size()];
   int i = 0;
   for(auto it=std::begin(executables);it!=std::end(executables);++it){
     exec_refs[i++] = exla::make<xla::LocalExecutable>(env, executables.at(i));
   }
+  // TODO: This should return the vector. There is an executable for every partition, usually 1.
```

Comment:
What is a partition in this case?

Reply:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/executable_build_options.h#L76

It's possible to partition operations between two devices...if you had a very large model and needed to split it, I think you could specify the number of partitions and then `Compile` takes care of partitioning the computation for you, and then you can run the executables you get back on different devices

Reply:
I see, thanks for explaining! So we would need to explicitly request it to split it, right? Maybe for now it will be good enough to just raise an error if it returns more than one entry? Something like:

```
if(execs.length > 1) {
  enif_raise_exception(env, enif_make_atom(env, "too_many_partitions"))
}
```

And then whenever, if ever, we need partitioning, we see how to tackle it. :)